### ISSUE TITLE
logging with global "x_values"

### ISSUE THREAD

**Issue by AntonioCarta** [ContinualAI/avalanche#375]
Metrics emit `<name, x, y>` values. Currently, `x` is updated inside the callback. Each callback can use its own `x` values. I propose to use a single global counter, incremented after every training iteratior. Basically, this global counter uniquely identify the model at a certain point in time. I think this is easier to implement and more intuitive (tensorflow does this automatically if I remember correctly). Also, now that we have the `eval` enabled every `k` training epochs, the `x` values depend on the frequency of the evaluation, which seems weird to me.

**Comment by AndreaCossu** [comment]
I agree on the need to have a global counter of this type. We should also take into consideration that some metrics are emitted at each iteration, other at each epoch, so the global counter should have the appropriate granularity.

**Comment by AntonioCarta** [comment]
I think a unique global counter is enough to handle all the others. The larger granularities (epochs, steps) will have larger offsets between emitted `x` values, but this is fine. I think having different counters may add complexity. Is there any advantage in having separate counters for each granularity?

In fact, the current `x` values do not add any information, because they are increased if and only if a value is emitted. If you ignore them the plot the result is the same.

**Comment by AndreaCossu** [comment]
I'm just wandering if the user could be confused from the fact that `AccuracyEpoch` has, let's say, x values of 1, 100, 200 instead of 1, 2, 3.

**Comment by vlomonaco** [comment]
Yes I think tensorflow does both: it has a internal counter but you can also set you own x value... maybe we can do the same? :)
